modules/andes-core/broker/src/main/java/org/wso2/andes/configuration/qpid/ServerConfiguration.java

```diff
@@ -340,24 +356,32 @@ public String interpolate(String obj) {
         }
     }
 
-    public static org.apache.commons.configuration.Configuration flatConfig(File file) throws ConfigurationException {
-        // We have to override the interpolate methods so that
-        // interpolation takes place accross the entirety of the
-        // composite configuration. Without doing this each
-        // configuration object only interpolates variables defined
-        // inside itself.
-        final MyConfiguration conf = new MyConfiguration();
-        conf.addConfiguration(new SystemConfiguration() {
-            protected String interpolate(String o) {
-                return conf.interpolate(o);
-            }
-        });
-        conf.addConfiguration(new XMLConfiguration(file) {
-            protected String interpolate(String o) {
-                return conf.interpolate(o);
-            }
-        });
-        return conf;
+    public static CombinedConfiguration flatConfig(File file) throws ConfigurationException {
```

Comment:
Shall we add a javadoc here?

Reply:
Added with https://github.com/wso2/andes/pull/1070/commits/86dd27a7538c822d25fb117f80e2b23bb60f600c